crates/worker/src/partition_processor_manager.rs

```diff
@@ -1092,11 +1107,21 @@ impl SpawnPartitionProcessorTask {
                         invoker.run(invoker_config),
                     )?;
 
-                    pp_builder
+                    if let Err(err) = pp_builder
                         .build::<ProtobufRawEntryCodec>(tc, bifrost, partition_store, configuration)
                         .await?
                         .run()
                         .await
+                    {
+                        let _ = events
+                            .send(ManagerEvent {
+                                partition_id,
+                                event: ProcessorEvent::Stopped(err),
```

Comment:
i would probably also return a `Stopped` event if the partition processor exited w/o an error. That way the receiver can also clean up its data structures.